lending/loan_management/doctype/loan_disbursement/loan_disbursement.py

```diff
@@ -165,8 +165,14 @@ def make_update_draft_schedule(self):
 
 		self.db_set("monthly_repayment_amount", schedule.monthly_repayment_amount)
 		if loan_details.status == "Sanctioned":
-			self.db_set("broken_period_interest", flt(schedule.broken_period_interest, precision))
-			self.db_set("broken_period_interest_days", flt(schedule.broken_period_interest_days, precision))
+			bpi_recovery_method = frappe.db.get_value(
+				"Loan Product", self.loan_product, "bpi_recovery_method"
+			)
+			if bpi_recovery_method == "Upfront Deduction":
```

Comment:
This value has to be set for every method. Add this check at the GL Level. Pass GL only if the method is "Upfront Deduction"